x-pack/plugins/fleet/server/services/package_policies/experimental_datastream_features.ts

```diff
@@ -74,6 +75,31 @@ export async function handleExperimentalDatastreamFeatureOptIn({
       // @ts-expect-error - TODO: Remove when ES client typings include support for synthetic source
       body,
     });
```

Comment:
We should probably do these whole block only if `existingOptIn?.features.synthetic_source` is defined no?

Reply:
I'm not sure, because the logic sets `_source.mode` even for `synthetic_source:false`. Though I can add an extra condition to only do this if `synthetic_source` changed.